controllers/datamovement_controller_lustre.go

```diff
@@ -263,9 +263,21 @@ func (r *DataMovementReconciler) createPersistentVolume(ctx context.Context, dm
 	for _, allocationSet := range storage.Spec.AllocationSets {
 		if allocationSet.TargetType == "MDT" || allocationSet.TargetType == "MGTMDT" {
 			fsName = allocationSet.FileSystemName
+			break
 		}
 	}
 
+	mgsNid := ""
```

Comment:
You can default this to `storage.Status.MgsNode` then get ride of lines 277-279.  Or keep the `if len(mgsNide) == 0` check and return an error - that ensures we don't create a PV with an empty mgsNid